lib/suma/processor.rb

```diff
@@ -28,53 +43,49 @@ def run(metanorma_yaml_path:, schemas_all_path:, compile:,
         exported_schema_config = collection_config.manifest.export_schema_config(schemas_all_path)
         exported_schema_config.path = schemas_all_path
 
-        Utils.log "Writing #{schemas_all_path}..."
         exported_schema_config.to_file
-        Utils.log "Done."
 
-        # now get rid of the source documents for schema sources
+        collection_config
+      end
 
+      def compile_schema(schemas_all_path, collection_config)
+        # now get rid of the source documents for schema sources
         col = Suma::SchemaCollection.new(
           config_yaml: schemas_all_path,
           manifest: collection_config.manifest,
           output_path_docs: "schema_docs",
           output_path_schemas: "plain_schemas",
         )
 
-        if compile
-          Utils.log "Compiling schema collection..."
-          col.compile
-        else
-          Utils.log "No compile option set. Skipping schema compilation."
+        col.compile
+      end
+
+      def compile_collection(collection_config, output_directory)
+        metanorma_collection, collection_opts = build_collection(collection_config, output_directory)
+
+        metanorma_collection.render(collection_opts)
+
+        # TODO: Temporarily disable removal of XML files
+        Dir.glob(File.join(Dir.getwd, output_directory,
+                           "*.xml")).each do |file|
+          puts "NOT DELETING ANY FILE #{file.inspect}"
+          # File.delete(file)
         end
+      end
 
+      def build_collection(collection_config, output_directory)
         new_collection_config_path = "collection-output.yaml"
         collection_config.manifest.remove_schemas_only_sources
         collection_config.to_file(new_collection_config_path)
 
-        if compile
```

Comment:
This is simplfying the flow a little too much.  I am happy to see spaghetti unravelled, but this was not my workflow, it was @koonwa and @ronaldtse , and I prefer them to weigh in on whether they need this code:

````
col = Suma::SchemaCollection.new(
           config_yaml: schemas_all_path,
           manifest: collection_config.manifest,
           output_path_docs: "schema_docs",
           output_path_schemas: "plain_schemas",
         )
````

and 

````
        new_collection_config_path = "collection-output.yaml"
         collection_config.manifest.remove_schemas_only_sources
         collection_config.to_file(new_collection_config_path)
````

to be run regardless of whether the `compile` variable is set, as is the case in the current code.

Reply:
Agreed, I really need a review from you @koonwa  and @ronaldtse

Reply:
@koonwa @ronaldtse any feedback appreciated